### PR TITLE
fix(devtools): fix the mapping between DOM tree and UI inspector

### PR DESCRIPTION
**Pull request by Cyunong**
Before submitting a new pull request, please make sure:

- [ ] Test cases have been added/updated/passed for the code you will submit.
- [ ] Documentation has added or updated.
- [ ] Commit message is following the [Convention Commit](https://conventionalcommits.org/) guideline with maximum 72 characters.
- [ ] Squash the repeat code commits, short patches are welcome.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=4122) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=4122) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=4122) it.</sub>

File: renderer/native/ios/renderer/component/view/HippyViewManager.mm [hunk 1]
```diff
@@ -178,6 +178,30 @@ - (void)measureInAppWindow:(NSNumber *)componentTag
...
+          callback(@[]);
+          return;
+      }
+      double locationX = [params[@"xOnScreen"] doubleValue];
+      double locationY = [params[@"yOnScreen"] doubleValue];
+      UIView* hitView = [rootView hitTest:{locationX, locationY} withEvent:nil];
+      NSDictionary *locationDict = @{
+          @"hippyTag": hitView.hippyTag,
+      };
+      callback(@[locationDict]);
```
Comment (wwwcg) [inline renderer/native/ios/renderer/component/view/HippyViewManager.mm:201]: May crash if hitView is nil or not a HippyView

File: renderer/native/ios/renderer/component/view/HippyViewManager.mm [hunk 2]
```diff
@@ -178,6 +178,30 @@ - (void)measureInAppWindow:(NSNumber *)componentTag
...
+          callback(@[]);
+          return;
+      }
+      UIView *rootView = viewRegistry[view.rootTag];
+      if (rootView == nil) {
+          callback(@[]);
+          return;
+      }
+      double locationX = [params[@"xOnScreen"] doubleValue];
+      double locationY = [params[@"yOnScreen"] doubleValue];
```
Comment (wwwcg) [inline renderer/native/ios/renderer/component/view/HippyViewManager.mm:196]: avoid using hardcode string

File: devtools/devtools-integration/native/include/devtools/devtools_utils.h [hunk 3]
```diff
@@ -55,8 +56,8 @@ class DevToolsUtil {
   static bool ShouldAvoidPostDomManagerTask(const std::string& event_name);
 
  private:
-  static std::shared_ptr<DomNode> GetHitNode(const std::shared_ptr<DomNode>& root_node, const std::shared_ptr<DomNode>& node, double x, double y);
-  static bool IsLocationHitNode(const std::shared_ptr<DomNode>& root_node, const std::shared_ptr<DomNode>& dom_node, double x, double y);
+    static hippy::dom::DomArgument makeLocationArgument(double x, double y);
```
Comment (wwwcg) [inline devtools/devtools-integration/native/include/devtools/devtools_utils.h:59]: indent issue

**Review by wwwcg — CHANGES_REQUESTED** [review]

File: renderer/native/ios/renderer/component/view/HippyViewManager.mm [hunk 4]
```diff
@@ -178,6 +183,33 @@ - (void)measureInAppWindow:(NSNumber *)componentTag
     }];
 }
 
+HIPPY_EXPORT_METHOD(getViewTagByLocation:(nonnull NSNumber *)componentTag
+                    params:(NSDictionary *__nonnull)params
+                    callback:(HippyPromiseResolveBlock)callback) {
+  [self.bridge.uiManager addUIBlock:^(__unused HippyUIManager *uiManager, NSDictionary<NSNumber *, UIView *> *viewRegistry) {
+      NSMutableDictionary *locationDict = [NSMutableDictionary dictionaryWithDictionary:@{
+        HippyTagKey: @(-1),
```
Comment (wwwcg) [inline renderer/native/ios/renderer/component/view/HippyViewManager.mm:191]: 这个-1 代表什么？要避免使用Magic number

File: renderer/native/ios/renderer/component/view/HippyViewManager.mm [hunk 5]
```diff
@@ -178,6 +183,33 @@ - (void)measureInAppWindow:(NSNumber *)componentTag
...
+          return;
+      }
+      UIView *rootView = viewRegistry[view.rootTag];
+      if (rootView == nil) {
+          callback(locationDict);
+          return;
+      }
+      double locationX = [params[HippyXOnScreenKey] doubleValue];
+      double locationY = [params[HippyYOnScreenKey] doubleValue];
+      UIView* hitView = [rootView hitTest:{locationX, locationY} withEvent:nil];
```
Comment (wwwcg) [inline renderer/native/ios/renderer/component/view/HippyViewManager.mm:205]: 如果找不到hitView表现是怎样的？

File: renderer/native/ios/renderer/component/view/HippyViewManager.mm (hunk 4 above)
Comment (Cyunong) [reply renderer/native/ios/renderer/component/view/HippyViewManager.mm:191]: -1是无效的tag，表示没有找到

File: renderer/native/ios/renderer/component/view/HippyViewManager.mm (hunk 5 above)
Comment (Cyunong) [reply renderer/native/ios/renderer/component/view/HippyViewManager.mm:205]: 找不到hitView返回的tag仍是-1，后续会直接使用rootView作为选中的View

**Review by wwwcg — DISMISSED** [review]

**Review by siguangli — APPROVED** [review]

**Review by zealotchen0 — APPROVED** [review]

**Pull request merged**